src/service.py

```diff
@@ -21,27 +22,51 @@
 
 engine = create_engine(DATABASE_URL)
 SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+bcrypt_context = CryptContext(schemes=["bcrypt"], deprecated="auto")
 
 
 def get_session() -> Generator[Session]:
     with SessionLocal() as session:
         yield session
 
 
-def get_project(project_id: uuid.UUID, db: Session) -> Projects | None:
-    return db.query(Projects).get(project_id)
+def get_project_with_user_access(db: Session, project_id: uuid.UUID, user_id: uuid.UUID) -> Projects | None:
+    query = select(Projects).join(UserProject).where(and_(Projects.id == project_id, UserProject.user_id == user_id))
+    return db.execute(query).scalar_one_or_none()
+
+
+def is_project_admin(db: Session, project_id: uuid.UUID, user_id: uuid.UUID) -> bool:
+    query = (
+        select(Projects)
+        .join(
+            UserProject,
+            (UserProject.project_id == Projects.id) & (UserProject.user_id == user_id) & (UserProject.is_admin == True),
+        )
+        .where(Projects.id == project_id)
+    )
+
+    project = db.execute(query).first()
+    if project is not None:
+        return True
+    else:
+        return False
```

Comment:
https://stackoverflow.com/questions/29161730/what-is-the-difference-between-one-and-first
Use one and except exception if you want to return boolean.
And the code you did can simplify to that
```suggestion
    return db.execute(query).first() is not None
```